pkg/kubeapiserver/authorizer/config.go

```diff
@@ -141,3 +158,37 @@ func (config Config) New() (authorizer.Authorizer, authorizer.RuleResolver, erro
 
 	return union.New(authorizers...), union.NewRuleResolvers(ruleResolvers...), nil
 }
+
+type ruleResolvedAUthorizer interface {
+	Authorize(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error)
+	RulesFor(user user.Info, namespace string) ([]authorizer.ResourceRuleInfo, []authorizer.NonResourceRuleInfo, bool, error)
+}
+
+type withNotSyncedReason struct {
+	ruleResolvedAUthorizer
+	informers []cache.SharedInformer
+}
+
+func (az withNotSyncedReason) Authorize(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+	d, reason, err := az.ruleResolvedAUthorizer.Authorize(ctx, a)
+	if err != nil {
+		return d, reason, err
+	}
+
+	if d == authorizer.DecisionNoOpinion && len(reason) == 0 && !az.hasSynced() {
+		reason = "informers not synced"
```

Comment:
adding this reason is misleading for the node authorizer for requests not made by nodes

Reply:
It does not claim that the decision not "no reason" when informers have finally synced. How do you see this as confusing?